packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts

```diff
@@ -364,21 +364,21 @@ describe("Client Factory generation", () => {
   });
 
   describe("should handle different auth options", () => {
-    it("should not generate credential if scope is empty", async () => {
+    it("should diagnost warning if scope is empty", async () => {
       try {
         await emitClientFactoryFromTypeSpec(
           `
-      @useAuth(
-        OAuth2Auth<[{
-          type: OAuth2FlowType.implicit,
-          authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
-          scopes: []
-        }]>)
-      @service( {title: "PetStoreClient"})
-      namespace PetStore;
-      `,
+          @useAuth(
+            OAuth2Auth<[{
+              type: OAuth2FlowType.implicit,
+              authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+              scopes: []
+            }]>)
+          @service( {title: "PetStoreClient"})
+          namespace PetStore;
+        `,
           false,
-          false
+          true
```

Comment:
yes, actually I could remove this default value, I explicitly set this because I want to differentiate these two cases, https://github.com/Azure/autorest.typescript/blob/4697bc247268d88e4ac2b870313adf7280c6e96b/packages/typespec-ts/test/unit/clientFactoryGenerator.spec.ts#L394